community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/CypherCompiler.scala

```diff
@@ -88,9 +83,11 @@ class CypherCompiler(graph: GraphDatabaseService,
 
   @throws(classOf[SyntaxException])
   def preParseQuery(queryText: String): PreParsedQuery = {
+    val logger = new RecordingNotificationLogger
```

Comment:
I don't understand where the execution mode affects the choice of logger here? Won't this mean that non-explain queries will report warnings?